### ISSUE TITLE
Support YouTube playlists

### ISSUE THREAD

**Issue by paulvt** [turt2live/matrix-dimension#347]
The YouTube widget currently does not seem to support playlist, even if you use an URL that points to a video but that also contains a list ID (e.g. https://www.youtube.com/watch?v=fooBarBaz&list=PL1_bla_bla_bla). The widget seems to disregard that ID and it is not possible to skip to the next/previous video similary to on YouTube itself.

Maybe there should be a different widget that you can feed just the list ID to make this work?

**Comment by TheTimeWalker** [comment]
So looking into this it seems like that it splits up the URL, takes the video URL and uses it to create an embed. I think a better idea would be to let it check if there is also a `list=` parameter. If yes, it would instead embed a playlist. This shouldn't be too difficult imo. That means we can keep the video widget intact and let the widget automatically detect if it's a video or playlist.

**Comment by TheTimeWalker** [comment]
Yeah it seems that the issue is the library [embed-video](https://github.com/alanshaw/embed-video) which is used for this widget doesn't have support for playlists. So either a new widget needs to be done for this - which imo is quite messy or the library has to be extended with that.

**Comment by turt2live** [comment]
Playlists aren't really supported by the spec for video widgets, so this would need to be a custom thing.

**Comment by paulvt** [comment]
Aww such a pity, was hoping it was just a matter of passing the list ID as well...
